### PR TITLE
Update bindgen to 0.64

### PR DESCRIPTION
**Pull request by asomers**
This fixes "reference to packed field is unaligned" errors from recent rustc.

### CODE REVIEW

File: pmc-rs/src/counter.rs [hunk 1]
```diff
@@ -360,9 +368,6 @@ fn init_pmc_once() -> Result<(), Error> {
             };
             return;
         }
-
-        // Register the signal handler
-        signal::watch_for(&[libc::SIGBUS, libc::SIGIO]);
```
Comment (domodwyer) [inline pmc-rs/src/counter.rs:365]: This signal handler is used in the [`read()` (and other) methods][read] to watch for the SIGBUS/SIGIO signals that the [libpmc docs] state:

> Applications using PMCs are required to handle the following signals:
> 
>      SIGBUS  When the hwpmc(4) module is unloaded using kldunload(8), pro-
>       cesses that have PMCs allocated to them will be sent a SIGBUS
>       signal.
> 
>      SIGIO   The hwpmc(4) driver will send a PMC owning process a SIGIO signal
>       if:
> 
>       o  If any process-mode PMC allocated by it loses all its target
>    		processes.
> 
>       o  If the driver encounters an error when writing log data to a
>   		configured log file.  This error may be retrieved by a subse-
>    		quent call to pmc_flush_logfile().

This follows the "Typical Program Flow" described in the same man page.

Your commit message for the change removing this mentions it was because you're seeing SIGIO during test execution, and that perhaps the test harness was the source of these; perhaps we disable SIGIO handling in tests if that's the case? What do you think?

[libpmc docs]:https://man.freebsd.org/cgi/man.cgi?query=pmc
[read]:https://github.com/domodwyer/pmc-rs/blob/master/pmc-rs/src/counter.rs#L286

File: pmc-rs/tests/pmc.rs [hunk 2]
```diff
@@ -35,7 +35,7 @@ fn test_set_counter() {
 fn test_counter_bad_name() {
     let err = CounterBuilder::default()
         .attach_to(vec![0])
-        .allocate("inst_retired.any")
```
Comment (domodwyer) [inline pmc-rs/tests/pmc.rs:38]: Huh, this is weird - maybe a bad copy/paste when I rewrote the public API?

File: pmc-rs/src/counter.rs (hunk 1 above)
Comment (asomers) [reply pmc-rs/src/counter.rs:365]: Unfortunately those signals can be caused by lots of reasons:
* SIGBUS is usually caused by an out-of-bounds memory access, and the chance that a program can recover is very low.  I don't think this library should try to catch this signal, because it could mask other errors in the program.
* SIGIO is normally sent by sockets with O_ASYNC set.  A pmc-rs consumer might legitimately want to use that signal, but it can't if the library registers its own signal handler.

The bottom line is that signal handlers need to be registered at the application level, not by a library, unless it's opt-in and explicit.  I don't think there's anything useful that the pmc-rs library can do with these signals that won't also potentially interfere with consuming applications.

File: pmc-rs/tests/pmc.rs (hunk 2 above)
Comment (asomers) [reply pmc-rs/tests/pmc.rs:38]: Yeah, probably.

**Comment by asomers** [comment]
CI is failing in the doc tests, which don't look like they ever worked.  I suggest merging this PR (after resolving the signal handler stuff) and fixing the doc tests later.

File: pmc-rs/src/counter.rs (hunk 1 above)
Comment (domodwyer) [reply pmc-rs/src/counter.rs:365]: Makes sense, I agree it's probably better for the application to register the handlers - I can add some docs that suggest it.

Probably worth `rm`-ing the whole signal module in that case :+1:

**Review by domodwyer — APPROVED** [review]
Thank you for this!

**Review by domodwyer — COMMENTED** [review]
Looks good in principle! The singal module needs `rm`-ing though.

**Comment by asomers** [comment]
Sorry @domodwyer this PR slipped away from me.  But I've now removed the signal module as you suggested.

**Comment by domodwyer** [comment]
No sorry is needed, thank you for the PR effort!

I pushed a couple commits to pass CI - just doc/lint fixes.

Thanks again!

**Review by domodwyer — APPROVED** [review]
